Sources/Datadog/Tracing/TracingAutoInstrumentation.swift

```diff
@@ -0,0 +1,103 @@
+    }
+
+    func apply() {
+        let interceptor = TracingRequestInterceptor.build(with: tracedHosts)
+        swizzler.swizzle(using: interceptor)
+    }
+}
+
+internal enum TracingRequestInterceptor {
+    static func build(with tracedHosts: Set<URL>) -> RequestInterceptor {
+        let interceptor: RequestInterceptor = { urlRequest in
+            guard let tracer = Global.sharedTracer as? DDTracer,
+                let someURL = urlRequest.url,
+                tracedHosts.allows(someURL),
+                HTTPHeadersWriter.canInject(to: urlRequest) else {
+                    return nil
+            }
+            let spanContext = tracer.createSpanContext()
+            let headersWriter = HTTPHeadersWriter()
+            headersWriter.inject(spanContext: spanContext)
+            let tracingHeaders = headersWriter.tracePropagationHTTPHeaders
+            var modifiedRequest = urlRequest
+            tracingHeaders.forEach { modifiedRequest.setValue($1, forHTTPHeaderField: $0) }
+
+            let operationName = someURL.absoluteString
+            let observer: TaskObserver = tracingTaskObserver(tracer: tracer, spanContext: spanContext, operationName: operationName)
+            return InterceptionResult(modifiedRequest: modifiedRequest, taskObserver: observer)
+        }
+        return interceptor
+    }
+
+    static func tracingTaskObserver(
+        tracer: DDTracer,
+        spanContext: DDSpanContext,
+        operationName: String
+    ) -> TaskObserver {
+        var startedSpan: OTSpan? = nil
+        let observer: TaskObserver = { observedEvent in
+            switch observedEvent {
+            case .starting(let request):
+                if let ongoingSpan = startedSpan {
+                    userLogger.warn("\(String(describing: request)) is starting a new trace but it's already started a trace before: \(ongoingSpan)")
+                    developerLogger?.warn("\(String(describing: request)) is starting a new trace but it's already started a trace before: \(ongoingSpan)")
+                }
+                let span = tracer.startSpan(
+                    spanContext: spanContext,
+                    operationName: operationName
+                )
+                let url = request?.url?.absoluteString ?? "unknown_url"
+                let method = request?.httpMethod ?? "unknown_method"
+                span.setTag(key: OTTags.httpUrl, value: url)
+                span.setTag(key: OTTags.httpMethod, value: method)
+                startedSpan = span
+            case .completed(let response, let error):
+                guard let completedSpan = startedSpan else {
+                    break
+                }
+                if let statusCode = (response as? HTTPURLResponse)?.statusCode {
+                    completedSpan.setTag(key: OTTags.httpStatusCode, value: statusCode)
+                }
+                if let someError = error {
+                    completedSpan.logError(someError)
+                }
```

Comment:
This is a critical part for trace rendering on https://app.datadoghq.com, but it's missing tests.